In [1]:
import spacy
from spacy.lang.en import English

In [2]:
# nlp.from_disk('../models')
nlp = spacy.load('../models/spacy_multi_cat_model/')

In [3]:
one_text = '''
Ok, let me say it again Come on, now you guys are just being piece of shit jews. I mean you have to admit, the guys in pink floyd play their instruments about as slow as a nigger works. I shouldn't even call what they play music. It's just a bunch of alarm clocks and cashier regirsters! But you know what the most pretentious thing about them is, its their lyrics. All af their songs are just surrealist poetry sung over doom-noise pop, and everyone starts calling them genius's over it. The truth is, their songs have no meaning. Take the album 'The Wall' for instance; sure it tells a story, but what is the moral and the meaning of the story? And dont tell me that the purpose of their songs is to make you think. The only way that music as slow as pink floyd could make you fucking think is if you were just as stoned as they are, which you kikes probly are... And one last time: 1) pink floyd fucking sucks 2) david fuckmor should taste my ass 3) you should to
'''



In [5]:
doc=nlp(one_text)
doc.cats

{'TOXIC': 0.9894261360168457,
 'SEVERE_TOXIC': 0.007514932658523321,
 'OBSCENE': 0.9430854916572571,
 'THREAT': 0.007477084174752235,
 'INSULT': 0.917741596698761,
 'IDENTITY_HATE': 0.7508668899536133}

## Before 5 epochs  
{'TOXIC': 0.9983283877372742,  
 'SEVERE_TOXIC': 0.0003475894336588681,  
 'OBSCENE': 0.2968767285346985,  
 'THREAT': 0.0011758265318349004,  
 'INSULT': 0.20628367364406586,  
 'IDENTITY_HATE': 0.00019311188952997327}  


## After 5 epochs  
{'TOXIC': 0.9834251999855042,  
 'SEVERE_TOXIC': 0.07075302302837372,  
 'OBSCENE': 0.8929150104522705,  
 'THREAT': 0.0113288052380085,  
 'INSULT': 0.8754300475120544,  
 'IDENTITY_HATE': 0.3847048282623291}  

## After 25 epochs  
  
{'TOXIC': 0.9894261360168457,  
 'SEVERE_TOXIC': 0.007514932658523321,  
 'OBSCENE': 0.9430854916572571,  
 'THREAT': 0.007477084174752235,  
 'INSULT': 0.917741596698761,  
 'IDENTITY_HATE': 0.7508668899536133}  